src/wallet/crypter.cpp

```diff
@@ -614,7 +612,7 @@ bool CCryptoKeyStore::EncryptKeys(CKeyingMaterial& vMasterKeyIn)
             if (!EncryptSecret(vMasterKeyIn, vchSecret, fvk.GetFingerprint(), vchCryptedSecret)) {
                 return false;
             }
-            if (!AddCryptedSaplingSpendingKey(fvk, vchCryptedSecret)) {
+            if (!AddCryptedSaplingSpendingKey(fvk, vchCryptedSecret, sk.DefaultAddress())) {
```

Comment:
Before we were defaulting to boost::none here. Could someone confirm that passing `.DefaultAddress()` is not incorrect here?

Reply:
@eirik0 This is part of EncryptKeys which is an experimental / developer only funtction.

Previously, with the default boost::none, the mapSaplingIncomingViewingKeys[] would not be set, but the function AddCryptedSaplingSpendingKey would still return true.

Now, mapSaplingIncomingViewingKeys[] does get set.  Is this the desired behaviour?

Reply:
`EncryptKeys` will only be calling `AddCryptedSaplingSpendingKey` for keys already in the wallet. So the mapping already exists and thus nothing needed to be set. Now, we'll just be adding it to the map again, which is fine, because it won't change the mapping or overwrite anything else.